### PR TITLE
tls: tls_cipher_suite keyword

### PR DESCRIPTION
**Pull request by fooinha**
- Allows tls app-layer rules to match TLS handshake cypher suites

With conjunction with  #2427, can be used, for instance, for detecting http/2 services allowing blacklisted tls cipher suites ->  https://http2.github.io/http2-spec/#BadCipherSuites.

Example rules:

alert tls any any -> any any (msg:"SURICATA TLS CIPHER SUITE CLIENT"; tls_cipher_suite:client:TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,0xcca9; sid:66666666; rev:1;)

alert tls any any -> any any (msg:"SURICATA TLS CIPHER SUITE SERVER"; tls_cipher_suite:server:0xcca8; sid:66666667; rev:1;)

### CODE REVIEW

**Comment by inliniac** [comment]
@thus can you have a look?

**Comment by inliniac** [comment]
The keyword looks old style (just looking at the PR title), @thus has been adding new keywords with a different notation.

**Comment by fooinha** [comment]
Is the current style using  '_'  as word separator?

**Comment by thus** [comment]
@inliniac: I'll try to find some time next week to review all the TLS PR's.

**Comment by thus** [comment]
@fooinha: Yup, that's right! The current style use '_' instead of '.' as a separator.

File: src/detect-tls-cipher-suite.c [hunk 1]
```diff
@@ -0,0 +1,1116 @@
...
+
+/**
+ * \file
+ *
+ * \author Paulo Pacheco <fooinha@gmail.com>
+ *
+ * Implements support for tls_cipher_suite keyword.
+ */
+
+#include "ctype.h"
```
Comment (thus) [inline src/detect-tls-cipher-suite.c]: This is already included in "suricata-common.h", which you include below.

File: src/detect-tls-cipher-suite.c [hunk 2]
```diff
@@ -0,0 +1,1116 @@
...
+
+/**
+ * \brief gets a cipher suite description
+ *
+ * \param value - 2 bytes value for cipher suite code
+ *
+ * \retval "-" if not found
+ * \retval const char * with description if found
+ */
+const char * TlsCipherSuiteDescription(uint16_t value)
```
Comment (thus) [inline src/detect-tls-cipher-suite.c]: I think everything related to cipher suite lookup should be moved to app-layer-ssl.[c|h], so we don't have to include detect-tls-cipher-suite.h everywhere we want to use the cipher suites.

**Comment by thus** [comment]
I haven't had time to take a good look at the detection code yet.

The code depends on custom tls logging code to be accepted (#2491 and #2449).

**Comment by thus** [comment]
Rebase is needed, btw.

**Comment by fooinha** [comment]
@thus, Thanx for review.
I changed the code in order to meet all your comments.
Also rebase was done.

**Comment by inliniac** [comment]
Please do new PR for incorporating feedback and/or rebases. Thanks!

**Pull request closed without merging**